### PR TITLE
Add versioning doc

### PR DESCRIPTION
**Pull request by Gnnng**

### CODE REVIEW

File: Versioning.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+### Master Branch
+
+Master branch always has the most active and latest version.
+
+### Release Branch
+
+There could be multiple release branches in the pattern of `release-<major>.<minor>`, such as `release-4.2`.
+
+The release branch always has the latest stable version for a certain minor version. However, some release versions may have already entered end-of-life, and the branches for them are only for achive purpose.
+
```
Comment (Revolution1) [inline Versioning.md:85]: maybe `<branch>-<commit>` or  `<pr-number>-<commit>` or `<major>.<minor>-<branch>` for pull request testing?

File: Versioning.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+### Master Branch
+
+Master branch always has the most active and latest version.
+
+### Release Branch
+
+There could be multiple release branches in the pattern of `release-<major>.<minor>`, such as `release-4.2`.
+
+The release branch always has the latest stable version for a certain minor version. However, some release versions may have already entered end-of-life, and the branches for them are only for achive purpose.
```
Comment (ansnunez) [inline Versioning.md]:
```suggestion
The release branch always has the latest stable version for a certain minor version. However, some release versions may have already entered end-of-life, and the branches for them are only for archiving purpose.
```

File: Versioning.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+- May or may not be compatible
+
+### Beta Pre-release
+
+Known:
+
+- New feature is stable
+- No Regression
+- Large-scale testing passed
+- Compatibility is confimred
```
Comment (ansnunez) [inline Versioning.md]:
```suggestion
- Compatibility is confirmed
```

File: Versioning.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+- Regression may happen on existing features
+- Large-scale testing may have issues
+- May or may not be compatible
+
+### Beta Pre-release
+
+Known:
+
+- New feature is stable
+- No Regression
```
Comment (ansnunez) [inline Versioning.md]:
```suggestion
- No regression
```

File: Versioning.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+Unknown:
+
+- Some undiscovered bug
+
+### Release
+
+Known:
+
+- Existing issues resolved
```
Comment (Revolution1) [inline Versioning.md:63]: Document is up to date.

File: Versioning.md (hunk 1 above)
Comment (Gnnng) [reply Versioning.md:85]: We don't really have a branch dedicated to PR testing purpose. Also, the branch name for a certain PR is usually chosen by the PR authors. So I'm not quite sure what are these names can be used for. Can you clarify a bit?

**Review by ansnunez — APPROVED** [review]

File: Versioning.md (hunk 1 above)
Comment (Revolution1) [reply Versioning.md:85]: Never mind.

**Review by ansnunez — APPROVED** [review]

**Review by Revolution1 — APPROVED** [review]

**Comment by Revolution1** [comment]
About image tagging:
add
- `<major>.<minor>` equal to the latest patch
- `<major>` equal to the latest minor
- `latest` equal to the latest stable version

**Comment by Gnnng** [comment]
> About image tagging:
> add
> 
> * `<major>.<minor>` equal to the latest patch
> * `<major>` equal to the latest minor
> * `latest` equal to the latest stable version

Thanks @Revolution1. These are quite good suggestions. We will revisit this doc and add more details about the image tagging once we have fully supported the those tags.

**Pull request merged**